prbt_hardware_support/test/unittests/unittest_brake_test_executor.cpp

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (c) 2019 Pilz GmbH & Co. KG
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <functional>
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+#include <ros/ros.h>
+
+#include <prbt_hardware_support/brake_test_executor_exception.h>
+#include <prbt_hardware_support/brake_test_executor.h>
+#include <prbt_hardware_support/BrakeTestErrorCodes.h>
+#include <prbt_hardware_support/joint_states_publisher_mock.h>
+
+namespace brake_test_executor_test
+{
+
+using namespace prbt_hardware_support;
+using namespace testing;
+
+static const std::string BRAKE_TEST_SERVICE_NAME{"/prbt/execute_braketest"};
+
+static const std::string BRAKETEST_ADAPTER_SERVICE_NAME{"/prbt/braketest_adapter_node/trigger_braketest"};
+
+class BrakeTestExecutorTest : public Test
+{
+public:
+  MOCK_METHOD2(triggerBrakeTest, bool(BrakeTest::Request&, BrakeTest::Response&));
+
+protected:
+  ros::NodeHandle nh_;
+};
+
+/**
+ * @brief Test execution of brake tests while robot is not moving.
+ *
+ * Test Sequence:
+ *  0. Setup Server for triggering the braketest
+ *  1. Set expectations and action on the service call
+ *  2. Publish fixed joint states.
+ *  3. Call brake test service.
+ *
+ * Expected Results:
+ *  0. Executor is create without problems, a client can attach to its service
+ *  1. -
+ *  2. -
+ *  3. Brake tests is triggered successfully. It is triggered exactly once.
+ */
+TEST_F(BrakeTestExecutorTest, testBrakeTestTriggeringRobotNotMoving)
+{
+  /**********
+   * Step 0 *
+   **********/
+  ros::ServiceServer service = nh_.advertiseService<BrakeTestExecutorTest, BrakeTest::Request, BrakeTest::Response>
+            (BRAKETEST_ADAPTER_SERVICE_NAME, &BrakeTestExecutorTest::triggerBrakeTest, this);
+
+  BrakeTestExecutor brake_test_executor(this->nh_);
+
+  ros::ServiceClient brake_test_srv_client_ = nh_.serviceClient<BrakeTest>(BRAKE_TEST_SERVICE_NAME);
+  ASSERT_TRUE(brake_test_srv_client_.exists()) << "Brake test service not available.";
```

Comment:
Can we be sure, that the service is available, or do we have to wait?

Reply:
The client should immediatly connect since the server is already constructed this should be fine...